lib/pf/task/api.pm

```diff
@@ -0,0 +1,82 @@
+package pf::task::api;
+
+=head1 NAME
+
+pf::task::api
+
+=cut
+
+=head1 DESCRIPTION
+
+pf::task::api
+
+=cut
+
+use strict;
+use warnings;
+use base 'pf::task';
+use POSIX;
+use pf::log;
+use pf::api;
+use pf::db;
+use threads;
+my $logger = get_logger();
+
+sub doTask {
+    my ($self, $args) = @_;
+    my ($method, @args) = @$args;
+    if (pf::api->isPublic($method)) {
+        my $pid;
+        if (pf::api->shouldFork($method)) {
+            pf::db::db_disconnect();
+            $pid = fork;
```

Comment:
just to make sure, do we get feedback in case the job fails when it forks when we fork it ?